src/mapeditdialog.cpp

```diff
@@ -1,100 +1,101 @@
 #include "mapeditdialog.hpp"
 #include "ui_mapeditdialog.h"
 
-#include "editor/map.hpp"
-
 MapEditDialog::MapEditDialog(
     const std::shared_ptr<const Editor::Project> project,
     std::shared_ptr<Misc::MapDocument> mapDocument,
-    QWidget *parent) : QDialog(parent),
-                       ui(new Ui::MapEditDialog),
-                       m_project(project)
+    QWidget *parent)
+    : QDialog(parent)
+    , m_ui(new Ui::MapEditDialog)
+    , m_project(project)
 {
-    ui->setupUi(this);
+    m_ui->setupUi(this);
 
     if (nullptr != mapDocument) {
         auto map(mapDocument->map());
-        ui->lineEditMapName->setText(mapDocument->mapName());
-        ui->lineEditChipset->setText(QString(map->chipset().c_str()));
-        ui->lineEditMusic->setText(QString(map->music().c_str()));
-        ui->spinBoxMapHeight->setValue(map->height());
-        ui->spinBoxMapWidth->setValue(map->width());
+        m_ui->lineEditMapName->setText(mapDocument->mapName());
+        m_ui->lineEditChipset->setText(QString::fromStdString(map->chipset()));
+        m_ui->lineEditMusic->setText(QString::fromStdString(map->music()));
+        m_ui->spinBoxMapHeight->setValue(map->height());
+        m_ui->spinBoxMapWidth->setValue(map->width());
     }
 }
 
 MapEditDialog::~MapEditDialog()
 {
-    delete ui;
+    delete m_ui;
 }
 
 QString MapEditDialog::getMapName() const {
-    return ui->lineEditMapName->text();
+    return m_ui->lineEditMapName->text();
 }
 
 quint16 MapEditDialog::getWidth() const {
-    return ui->spinBoxMapWidth->value();
+    return m_ui->spinBoxMapWidth->value();
 }
 
 quint16 MapEditDialog::getHeight() const {
-    return ui->spinBoxMapHeight->value();
+    return m_ui->spinBoxMapHeight->value();
 }
 
 QString MapEditDialog::getChipset() const {
-    return ui->lineEditChipset->text();
+    return m_ui->lineEditChipset->text();
 }
 
 QString MapEditDialog::getMusic() const {
-    return ui->lineEditMusic->text();
+    return m_ui->lineEditMusic->text();
 }
 
 void MapEditDialog::onChipsetBrowse() {
-    QString chipsetPath(
-        (m_project->coreProject().projectPath() / "chipsets").string().c_str()
-    );
+    QString chipsetPath = QString::fromStdString(
+                (m_project->coreProject().projectPath() / "chipsets")
+                .string());
+
     // On Windows, fs::pash puts some backslashes. That sucks.
-    chipsetPath.replace("\\", "/");
+    // Clean path uses slashes, remove weird paths as "folder/../folder"
+    chipsetPath = QDir::cleanPath(chipsetPath);
+
     QFileDialog dlg(this, tr("Choose the chipset file for your map."),
                     chipsetPath, "PNG files (*.png)");
     dlg.exec();
 
-    if(dlg.result() == QDialog::Accepted) {
-        int index;
+    if (dlg.result() == QDialog::Accepted) {
         QStringList selectedFiles = dlg.selectedFiles();
         QString selectedChipset = selectedFiles.at(0);
-        if((index = selectedChipset.indexOf(chipsetPath)) < 0) {
```

Comment:
Attributions in test is not a good idea. Let's do that one line before to be sure.

Reason : if you add another condition as : 

```if (shouldWeDoTheTestBool && (index = selectedChipset.indexOf(chipsetPath)) < 0)```
And shouldWeDoTheTestBool = false, the attribution will not even be done.